github/snapcraft.py

```diff
@@ -1,37 +1,122 @@
-#!/usr/bin/env python2
+#!/usr/bin/python2
 
-import urllib2
+import requests
 import urlparse
 
+import re
 
-def get_plugin_count(github_repo_url):
-    count = {}
-    repo = _get_repo_from_url(github_repo_url)
-    snapcraft_yaml_file = _get_snapcraft_yaml_file(repo)
-    if snapcraft_yaml_file:
-        for line in snapcraft_yaml_file:
-            line = line.strip()
-            if line.startswith('plugin: '):
-                plugin = line[8:]
-                if plugin in count:
-                    count[plugin] = count[plugin] + 1
-                else:
-                    count[plugin] = 1
-        snapcraft_yaml_file.close()
-    return count
+_scriplets = ["prepare", "build", "install", "version-script"]
+_tar_type_regex = re.compile(r'.*\.((tar(\.(xz|gz|bz2))?)|tgz)$')
 
 
 def _get_repo_from_url(github_repo_url):
     parsed_url = urlparse.urlparse(github_repo_url)
     return parsed_url.path[1:]
 
 
-def _get_snapcraft_yaml_file(github_repo):
+def get_snapcraft_yaml_file(github_repo_url):
+    """
+    Extracts snapcraft file from reposutory
+    :param github_repo_url: string, url of the repository
+    :return:
+    """
+
+    github_repo = _get_repo_from_url(github_repo_url)
     paths = ['snapcraft.yaml', '.snapcraft.yaml', 'snap/snapcraft.yaml']
     for path in paths:
         url = 'https://raw.githubusercontent.com/{}/master/{}'.format(
             github_repo, path)
-        try:
-            return urllib2.urlopen(url)
-        except:
-            pass
+        resp = requests.request("GET", url)
```

Comment:
I prefer to use full words, because abbreviations are complex for non-native english speakers, like me. Please replace `resp` with `response`.